feat(fileOps): add copy, rename, and delete operations for files and …

taralamia opened the pull request:
This PR introduces core file and folder operations to the storage system, including:

 Delete: Remove files or folders along with nested contents

 Copy: Duplicate files or folders, preserving metadata

 Rename: Update file or folder names while maintaining path consistency

taralamia committed:
feat(fileOps): add copy, rename, and delete operations for files and folders

parvez-ahammed commented on controllers/fileOpsController.js:
```diff
@@ -0,0 +1,78 @@
+  } catch (err) {
+    next(err);
+  }
+}
```
nitpick: you can keep one empty line between two method to make the code more consistent

parvez-ahammed commented on controllers/storageController.js:
```diff
@@ -36,9 +36,47 @@ async function uploadPdf(req, res, next) {
+    const userId = req.user._id;
+
+    if (!itemId || !itemType) {
+      return next(createError(400, 'itemId and itemType are required'));
```
Future updates: explore validation library like ZOD which can be used to do this kind of work using middlewares. So this kind of validation check is expected to be handled by middlewares. Good for now but there can be improvements.

parvez-ahammed commented on routes/route.js:
```diff
@@ -10,6 +10,8 @@ const googleRoutes = require('./auth/googleRoutes');
 const folderRoutes = require('./storage/folderRoutes');
 const infoRoutes = require('./storage/infoRoutes');
 const fileRoutes = require('./storage/fileRoutes');
+
```
extra whitespace

parvez-ahammed commented on services/fileOpsService.js:
```diff
@@ -0,0 +1,223 @@
+    const baseName = path
+      .basename(file.fileName, path.extname(file.fileName))
+      .toLowerCase()
+      .replace(/\s+/g, '-');
```
Maybe this code is duplicated. Any duplicate could should be transformed into a method if possible.

taralamia committed:
refactor: improve folder copy and rename handling
- Added recursive traversal for copying nested folders and files
- Updated folder renaming to also rename nested file and folder paths
- Extracted file-matching logic to helper to follow DRY principle
